### PR TITLE
Bundle limitations and gotchas together in the README.md

### PR DESCRIPTION
**Pull request by spawnia**
The documentation of this action is really good and has a lot of information. We can make it easier to use by putting a common example first, then clearly listing the limitations and possible gotchas.

I found that there was a bit of duplication, so i attempted to consolidate and distill it down to the essentials.

Thank you for this package!

### CODE REVIEW

**Comment by spawnia** [comment]
Is there any downside to always defining `${{ github.head_ref }}`? I think we might save even more people the trouble if we just say it's required.

**Comment by stefanzweifel** [comment]
Thanks for this PR! 🙌 

> Is there any downside to always defining ${{ github.head_ref }}? I think we might save even more people the trouble if we just say it's required.

There is. `github.head_ref` is only available when the Action is used using the `pull_request` event. That's why we can't use it as a default value.

File: README.md [hunk 1]
```diff
@@ -41,34 +36,27 @@ Add the following step at the end of your job.
...
 
 ```yaml
 name: php-cs-fixer
 
-on: pull_request
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
```
Comment (stefanzweifel) [inline README.md:59]: Did this example work? Usually listening to both pull_request and push events causes issues with the Action. 
I assume it worked, as the push event is only triggered on `master`?

Comment (spawnia) [reply README.md:59]: Just to be sure, i tried this config in a project of mine.

Here is the successful run on master: https://github.com/spawnia/sailor/runs/650648517

And here is what happens on a PR: https://github.com/spawnia/sailor/runs/650660575

**Comment by spawnia** [comment]
> There is. github.head_ref is only available when the Action is used using the pull_request event. That's why we can't use it as a default value.

It does not seem to error when trying to get its value, so the result is that you just get an empty string - which is what the current default is, if i am not mistaken.

I actually just tried running it directly on a branch with `${{ github.head_ref}}` and it worked perfectly: https://github.com/spawnia/sailor/runs/650680073

File: README.md (hunk 1 above)
Comment (stefanzweifel) [reply README.md:59]: This is great. Thanks!

**Comment by stefanzweifel** [comment]
Hm, seems this has changed since I tested this the last time. 😅

If this actually works, I think it would even make sense to use `github.head_ref` as the default value for `branch`. (Will do some testing in the next days)

**Comment by stefanzweifel** [comment]
Using `github.head_ref` as a default value actually works. See #75 
I've tested this in a test project while listening to `push` and `pull_request`-events.

One downside: For `pull_request`-events the `checkouts`-Action still has to be updated with a `ref`-input. Otherwise the repo would be in a detached state.

Still thinking about if this would be a good approach. Would merge those PRs and update the README with an updated example, if I think we should go down that path.

**Comment by spawnia** [comment]
@stefanzweifel i simplified the README further and clarified the `branch` option is only needed to push to other branches.

**Comment by stefanzweifel** [comment]
Just wanted to update the README too. Will do some last testing and then tag 4.2.0

**Comment by spawnia** [comment]
@stefanzweifel feel free to edit my branch as needed or pick the parts you like :(

**Pull request merged**